java/org/apache/catalina/connector/Response.java

```diff
@@ -1197,16 +1197,12 @@ public String encodeUrl(String url) {
     public void sendAcknowledgement()
         throws IOException {
 
-        if (isCommitted()) {
```

Comment:
The move to `sendAcknowledgement()` changes the behavior, doesn't it?

Reply:
The `isCommitted` check moved into `Coyote.Request` so the behavior should be the same.

Reply:
I'm still uncomfortable with this check moving. I like that it is the first check in all the `connector.Reponse` methods that need it. I think I'd prefer duplication (as per `setCharacterEncoding()`).

Reply:
I've put the `isCommitted` check back into `catalina.connector.Response`